src/main/java/duke/DateString.java

```diff
@@ -0,0 +1,46 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+
+/**
+ * Represents dates used by the Deadline and Event classes. Internally contains a
+ * java.time.LocalDate object that outputs the date in a standardised pattern,
+ * but silently falls back to using a string if the input cannot be parsed.
+ */
+public class DateString {
+    private final String dateString;
+    private LocalDate localDate;
+
+    /**
+     * Constructor which takes in a string and attempts to parse it to create a LocalDate
+     * object. If the parsing is unsuccessful, silently defaults to using a string instead.
+     * @param str
+     */
+    public DateString(String str) {
+        this.dateString = str;
```

Comment:
An advanced coding standard that you can consider following is to avoid unnecessary use of `this` with fields. The `this` keyword should only be used when a field is shadowed by a method or constructor parameter. You can refer to the standards [here](https://se-education.org/guides/conventions/java/index.html)